### ISSUE TITLE
Ext API roles ROLE_API_GROUPS_* not working as supposed

### ISSUE THREAD

**Issue by mliradelc** [opencast/opencast#1097]
**Describe the bug**
If you have a user with the roles ROLE_API_GROUPS_* (VIEW, CREATE, EDIT, DELETE) you can't see anything through the API with this user. 

**To Reproduce**
1. Create (Or use) an user with the mentioned roles.
2. Try to get, post o delete a group through the API using the user with these roles.
    1. For example GET https://develop.opencast.org/api/groups?&limit=0&offset=0 With the authentication of the user
    3. It will in:
        * Version 7.x a 401 error.
        * Version 8.x a 200 status but a blank page.
4. Trying with the admin user, shows the information without problem.

**Environment Information**
 - OS: CentOS 7
 - Browser Safar
 - Opencast version 7.x and 8.x

**Comment by staubesv** [comment]
@mliradelc The reason is that groups themselves don't have any ACLs and only ROLE_ADMIN and Tenant Admins have access to groups in Opencast. Up to some part, this makes sense because groups having no ACLs implies that one can either see all groups or none. IIRC, we have a custom workaround in our codebase that allows an API user with ROLE_API_GROUPS_VIEW (or similiar) to access all groups. I can post some more details here next week.

**Comment by mliradelc** [comment]
> I can post some more details here next week.

@staubesv Yes, please 🙏! 

I don't understand for what is this Role if you can't use it to allow access to the group's ext API, so what is the functionality

**Comment by gregorydlogan** [comment]
IMO, this is a bug in 8.x.  While I personally disagree with the restrictions placed on `ROLE_API_GROUPS_*`, I can see the technical reasoning behind it.  Considering those limitations, returning a 401 makes a *lot* more sense than returning a 200 with nothing to see.

**Comment by mliradelc** [comment]
Here I'm a little confused about how should be work, because If the Role is superfluous, how this endpoint should be accessed?

**Comment by staubesv** [comment]
@gregorydlogan AFAIK, the behaviour has always been like that and was not introduced by OC 8. 401 would certainly be a better choice here.

@mliradelc There is one use can: The tenant admin role can access the group but cannot access the API endpoint in the default configuration (ROLE_ADMIN can).

**Comment by staubesv** [comment]
@mliradelc Here is what we currently do: Essentially, we removed the logic that hides groups from non-admins from the index service but added checks to the external API.
Obviously, this solution does not cover "every relevant place" in Opencast - That is why you never saw a pull request.

GroupsEndpoint.java

```
  private static final String ROLE_CREATE = "ROLE_API_GROUPS_CREATE";
  private static final String ROLE_DELETE = "ROLE_API_GROUPS_DELETE";
  private static final String ROLE_EDIT = "ROLE_API_GROUPS_EDIT";
  private static final String ROLE_VIEW = "ROLE_API_GROUPS_VIEW";

  [....]

  private SecurityService securityService;

  public void setSecurityService(SecurityService securityService) {
    this.securityService = securityService;
  }

  [...]

 [... for all affected requests, we do this: ...]
  if (!isUserAuthorized(ROLE_VIEW)) {
     return ApiResponses.Json.ok(ApiVersion.VERSION_1_0_0, arr(new ArrayList<JValue>()));
  }


 protected boolean isUserAuthorized(String... roles) {
    User user = securityService.getUser();
    Organization org = securityService.getOrganization();
    List<String> authorizedRoles = new ArrayList<>();
    authorizedRoles.add(SecurityConstants.GLOBAL_ADMIN_ROLE);
    authorizedRoles.add(org.getAdminRole());
    authorizedRoles.addAll(Arrays.asList(roles));

    boolean isUserAuthorized = false;
    for (String authorizedRole : authorizedRoles) {
      if (user.hasRole(authorizedRole)) {
        isUserAuthorized = true;
        break;
      }
    }
    return isUserAuthorized;
  }
```

GroupQueryBuilder.java (in the module index-service)

We removed the following codeblock from the method buildQuery (approx. line 79):

```
  if (query.getActions() != null && query.getActions().length > 0) {
      User user = query.getUser();
      if (!user.hasRole(GLOBAL_ADMIN_ROLE) && !user.hasRole(user.getOrganization().getAdminRole())) {
        for (Role role : user.getRoles()) {
          for (String action : query.getActions()) {
            and(EventIndexSchema.ACL_PERMISSION_PREFIX.concat(action), role.getName());
          }
        }
      }
    }

```

**Comment by mliradelc** [comment]
> @mliradelc Here is what we currently do: Essentially, we removed the logic that hides groups from non-admins from the index service but added checks to the external API.
> Obviously, this solution does not cover "every relevant place" in Opencast - That is why you never saw a pull request.
> 
>
>   [....]
> 

Hi, @staubesv  worked like a Charm, now the Role works as Stated, It would be good to have this Role working as standard in Opencast, Still I don't see where is the security problem, is the Administrator who allows what can one see or do.

I had to adjust a few libraries, modify one or two lines and I think that I found a bug in the checkstyle script. It works very well. Now I will try with the Ilias plugging this. 😄 


PD: Now in the tests fails the Animate module but here we don't use that, so It's not an issue for me.


Edit: I sang victory very Earlier, POST is unauthorized still, now that I looked the code, the ROLE_VIEW is only authorized
```java
if (!isUserAuthorized(ROLE_VIEW)) {
     return ApiResponses.Json.ok(ApiVersion.VERSION_1_0_0, arr(new ArrayList<JValue>()));
  }
```
Should be for the others the same right?
